azure-pipelines/publishing/templates/release-docker-build-steps.yml

```diff
@@ -0,0 +1,74 @@
+# Template: Steps to build all components for a specific OS
+# Consume this steps template in one or more jobs by passing in parameter values.
+
+parameters:
+- name: targetOsArch        # example: debian9_arm32
+  type: string
+- name: imageVersion
+  type: string
+
+
+jobs:
+- job: ${{parameters.targetOsArch}}
+  steps:
+  - checkout: self
```

Comment:
>checkout [](http://example.com/codeflow?start=4&length=8)

Do we need the checkout step at all? Isn't 'self' the default?

Reply:
That's what I thought too - but after templatization having nested templates was causing issues. The 2nd stage in the github-release.yml wasn't performing checkout unless explicitly specified. It's unclear if it's a bug or a feature, though I did see some people posting about similar issues online, but they look a bit dated:

[https://developercommunity.visualstudio.com/t/azure-devops-pipeline-with-template-does-not-check/579283](https://developercommunity.visualstudio.com/t/azure-devops-pipeline-with-template-does-not-check/579283)